packages/mui-base/src/useSelect/useSelect.ts

```diff
@@ -297,27 +306,20 @@ function useSelect<OptionValue, Multiple extends boolean = false>(
     rootRef: mergedListRootRef,
   } = useList(useListParameters);
 
-  const createHandleButtonMouseDown =
-    (externalEventHandlers?: EventHandlers) => (event: React.MouseEvent & MuiCancellableEvent) => {
-      externalEventHandlers?.onMouseDown?.(event);
-      if (!event.defaultMuiPrevented) {
-        const action: ButtonClickAction = {
-          type: SelectActionTypes.buttonClick,
-          event,
-        };
-
-        dispatch(action);
-      }
-    };
+  // store the initial open state to prevent focus stealing
+  // (the first option gets focused only when the menu is opened by the user)
+  const isInitiallyOpen = React.useRef(open);
 
   useEnhancedEffect(() => {
-    // Scroll to the currently highlighted option.
-    if (highlightedOption != null) {
+    if (open && highlightedOption !== null && !isInitiallyOpen.current) {
       const optionRef = getOptionByValue(highlightedOption)?.ref;
+
       if (!listboxRef.current || !optionRef?.current) {
         return;
       }
 
+      optionRef.current.focus({ preventScroll: true });
```

Comment:
We could rely on the native scroll into view for focused elements, but the result is slightly different (by ~25px), so I kept the original one.

What do you think @zanivan? Is it worth it to keep our custom scroll behavior or is it ok to rely on the slightly different native one?

Reply:
Thanks for the videos—I must admit that it was hard to visualize what was going on without this 😅 

I think we can go with the native scroll, however the layout shift concerns me a bit. Using the videos by example, in the first one, it's way easier to understand where you're scrolling to. But again, I don't see this as a blocker since it's a native behavior.

Reply:
> Using the videos by example, in the first one, it's way easier to understand where you're scrolling to

Yeah. Also, I've kept experimenting, and with native scrolling, there's a difference between browsers. It might be good to have a consistent scrolling behavior across browsers 🤔

Reply:
Yep, thought the same—maybe in this case the best code will be more code then? 🤣